nuspec/chocolatey/chocolatey/tools/chocolateysetup.psm1

```diff
@@ -735,58 +736,59 @@ if (Test-Path($ChocolateyProfile)) {
-    Write-Output "'$env:SystemRoot\Microsoft.Net\$NetFxArch\v4.0.30319' was not found or this is forced"
-    if (!(Test-Path $NetFx4Path)) {
-      Write-Output "Creating folder `'$NetFx4Path`'"
-      $null = New-Item -Path "$NetFx4Path" -ItemType Directory
+  $NetFx48Url = 'https://download.visualstudio.microsoft.com/download/pr/2d6bb6b2-226a-4baa-bdec-798822606ff1/8494001c276a4b96804cde7829c04d7f/ndp48-x86-x64-allos-enu.exe'
+  $NetFx48Path = "$tempDir"
+  $NetFx48InstallerFile = 'ndp48-x86-x64-allos-enu.exe'
+  $NetFx48Installer = Join-Path $NetFx48Path $NetFx48InstallerFile
+
+  if (((Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full" -ErrorAction SilentlyContinue).Release -lt 528040) -or $forceFxInstall) {
+    Write-Output "The registry key for .Net 4.8 was not found or this is forced"
+
+    if (Test-Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Component Based Servicing\RebootPending") {
+        Write-Warning "A reboot is required. `n If you encounter errors, reboot the system and attempt the operation again"
     }
 
-    $netFx4InstallTries += 1
+    $netFx48InstallTries += 1
 
-    if (!(Test-Path $NetFx4Installer)) {
-      Write-Output "Downloading `'$NetFx4Url`' to `'$NetFx4Installer`' - the installer is 40+ MBs, so this could take a while on a slow connection."
-      (New-Object Net.WebClient).DownloadFile("$NetFx4Url","$NetFx4Installer")
+    if (!(Test-Path $NetFx48Installer)) {
+      Write-Output "Downloading `'$NetFx48Url`' to `'$NetFx48Installer`' - the installer is 100+ MBs, so this could take a while on a slow connection."
+      (New-Object Net.WebClient).DownloadFile("$NetFx48Url","$NetFx48Installer")
     }
 
     $psi = New-Object System.Diagnostics.ProcessStartInfo
-    $psi.WorkingDirectory = "$NetFx4Path"
-    $psi.FileName = "$NetFx4InstallerFile"
+    $psi.WorkingDirectory = "$NetFx48Path"
+    $psi.FileName = "$NetFx48InstallerFile"
     # https://msdn.microsoft.com/library/ee942965(v=VS.100).aspx#command_line_options
     # http://blogs.msdn.com/b/astebner/archive/2010/05/12/10011664.aspx
     # For the actual setup.exe (if you want to unpack first) - /repair /x86 /x64 /ia64 /parameterfolder Client /q /norestart
-    $psi.Arguments = "/q /norestart /repair"
+    $psi.Arguments = "/q /norestart"
 
-    Write-Output "Installing `'$NetFx4Installer`' - this may take awhile with no output."
+    Write-Output "Installing `'$NetFx48Installer`' - this may take awhile with no output."
     $s = [System.Diagnostics.Process]::Start($psi);
     $s.WaitForExit();
-    if ($s.ExitCode -ne 0 -and $s.ExitCode -ne 3010) {
-      if ($netFx4InstallTries -ge 2) {
+    if ($s.ExitCode -eq 1641 -or $s.ExitCode -eq 3010) {
+      throw ".NET Framework 4.8 was installed, but a reboot is required. `n Please reboot the system and try to install/upgrade Chocolatey again."
```

Comment:
What happens at this point, if this is hit?

If they run the install.ps1 again, it will likely fail, and the `c:\programdata\chocolatey` folder will already exist, right?

Reply:
Ignore this! Just seen your comment in the description on this PR.